pkg/api/handlers/mcpgateway/clientmessagehandler.go

```diff
@@ -185,6 +185,7 @@ func (c *clientMessageHandler) handleMessage(ctx context.Context, msg nmcp.Messa
 
 	var ch <-chan nmcp.Message
 	if msg.ID != nil {
+		msg.ID = uuid.NewString()
```

Comment:
I know we talked about this, but a comment explaining why we are doing this would be nice here.